### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by imubarak234**
- Create a class Classroom with the following: 
- - @label instance variable, should be initialized in the constructor.
- - Setter and getter for @label (remember about attr_accessor).
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
- - Create the has-many side (a classroom has many students).
- - Create the belongs-to side (a student belongs to a classroom).
- - Make sure that when adding a student to a classroom it also sets the classroom for the student.
- - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class Book with the following:
- - @title and @author instance variables, should be initialized in the constructor.
- - Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following:
- - @date instance variable, should be initialized in the constructor.
- - Setter and getter for @date (remember about attr_accessor).
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- - Create the has-many side of Book and Rental (a book has many rentals).
- - Create the belongs-to side of Rental and Book (a rental belongs to a book).
- - Create the has-many side of Person and Rental (a person has many rentals).
- - Create the belongs-to side of Rental and Person (a rental belongs to a person).
- - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
+
+  attr_accessor :author, :title
+  attr_reader :rental
+end
```
Comment (er-contreras) [inline book.rb]: - [ ] A method for adding rentals should be present. You could use something like this:

```
  def add_rental(person, date) 
    Rental.new(date, self, person)
  end
```

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    @book = book
+    book.rental << self
+
+    @person = person
+    person.rental << self
+  end
+
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (er-contreras) [inline rental.rb]: - [ ] `Getters and Setters for @book and @person instance variables should be present`

File: person.rb [hunk 3]
```diff
@@ -7,9 +7,10 @@ def initialize(age, name = 'unknown', parent_permission: true)
     @age = age
     @permission = parent_permission
     @correct = Corrector.new
+    @rental = []
   end
 
-  attr_reader :id, :name, :age, :permission
+  attr_reader :id, :name, :age, :permission, :rental
```
Comment (er-contreras) [inline person.rb:14]: - [ ] `A method for adding rentals should be present` something as follows:

```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
end
```

**Review by er-contreras — CHANGES_REQUESTED** [review]
Hi @imubarak234, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

almost there

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ignatius22 — APPROVED** [review]
Hi @imubarak234 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**